### DISCUSSION TITLE
Is it possible to input a file and get the transformed contents out on stdout

### DISCUSSION THREAD

**Discussion by kaleidawave** [biomejs/gritql#432] [General]
Hi! Have been playing with gritql a bit and made a query that finds a Rust enum and produces a match statement. (In a sense imitating the rust-analyser 'fill match branches' but on the command line without the LSP).

The query is
```gritql
language rust

enum_item(name = $tn, body=enum_variant_list(variants=$v)) where {
	// $tn :< $name
    $branches = [],
    $v <: some bubble($tn, $branches) or {
        enum_variant($name, body=ordered_field_declaration_list(type=$ttype)),
        enum_variant($name, body=field_declaration_list(declarations=$ftype)),
        enum_variant($name)
    } where {
        if ($ttype <: not undefined) {
            $fields = [],
            $ttype <: some bubble($fields) `$n` where {
                $fields += `_`
            },
            $f = join(list=$fields, separator=","),
            $branches += `$tn::$name($f) => {}`
        } else if ($ftype <: not undefined) {
            $fields = [],
            $ftype <: some bubble($fields) field_declaration($name) where {
                $fields += `$name`
            },
            $f = join(list=$fields, separator=", "),
            $branches += `$tn::$name { $f } => {}`
        } else {
            $branches += `$tn::$name => {}`
        }
    },
    $new = join(list=$branches, separator=",
    ")
} => `match _ {
    $new
}`
```

I think I can save this as a reusable pattern inside the `.grit` directory. The part I am not quite sure exists is whether a command exists in the grit CLI that I can run that takes a file and produces the match result to stdout. I understand the linting `grit check` and the general file mutations `grit apply` (akin to codemod), what I want it something that just pipes a string in and results matched transformed results out. Something like `grit find my_file.rs --pattern 'enum_to_match(name="MyEnum")' --stdout`?

Thanks 🙏

**Comment by morgante** [comment]
The `--stdin` flag will do this:

```
cat myfile.rs | grit apply 'enum_to_match(name="MyEnum")' --stdin --lang=rust myfile.rs
```

**Reply by morgante** [reply]
Also, if you want to apply against a file and see the output without editing the file on disk you can just use the `--dry-run` flag:

```
grit apply 'enum_to_match(name="MyEnum")' --dry-run --lang=rust myfile.rs
```

**Reply by kaleidawave** [reply]
Thanks that works great!
